preconditions/src/main/java/com/palantir/logsafe/Preconditions.java

```diff
@@ -0,0 +1,262 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ * Adapted from Guava 21.0 under the Apache 2 license; original copyright below.
+ */
+
+/*
+ * Copyright (C) 2007 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.palantir.logsafe;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import com.palantir.logsafe.exceptions.SafeNullPointerException;
+
+public final class Preconditions {
+    private Preconditions() {}
+
+    /**
+     * Ensures the truth of an expression involving one or more parameters to the calling method.
+     *
+     * @param expression a boolean expression
+     * @throws SafeIllegalArgumentException if {@code expression} is false
+     */
+    public static void checkArgument(boolean expression) {
+        if (!expression) {
+            throw new SafeIllegalArgumentException();
+        }
+    }
+
+    /**
+     * Ensures the truth of an expression involving one or more parameters to the calling method.
+     *
+     * @param expression a boolean expression
+     * @param message the loggable exception message
+     * @throws SafeIllegalArgumentException if {@code expression} is false
+     */
+    public static void checkArgument(boolean expression, String message) {
+        if (!expression) {
+            throw new SafeIllegalArgumentException(message);
+        }
+    }
+
+    /**
+     * Ensures the truth of an expression involving one or more parameters to the calling method.
+     *
+     * <p>See {@link #checkArgument(boolean, String, Arg...)} for details.
+     */
+    public static void checkArgument(boolean expression, String message, Arg<?> arg) {
```

Comment:
do we need the 1-3 arg variations? aren't they covered by the varargs method?

if https://stackoverflow.com/a/3737976 is still relevant, then let's just use SafeVarargs

Reply:
I believe the reason is mostly perf, which is guava Preconditions reason for defining additional methods.  Without it you always create an array whenever you call a precondition (which is a lot). With it, you only make an array if the precondition fails (which shouldn't be that much).

Reply:
I think there's a miniscule performance improvement because you don't have to allocate an array.

Reply:
Happy to get rid of them, but I figured this class itself is unlikely to change much (so the worse readability isn't likely to matter than much), and I was hoping this would preferably not have too much worse performance compared to guava.

Reply:
Eh I just spotted that all the exception constructors use varags too, so I'm not sure we really gain anything.  Gonna delete to get the minimal API in.

Reply:
The difference is you are hopefully throwing exceptions very rarely, whereas you are usually calling the Preconditions very frequently

Reply:
Watch out for subtle API breaks like the one introduced around guava 19. I'd prefer to be on the safe side now and copy exactly what guava does.